src/slave/containerizer/mesos/paths.hpp

```diff
@@ -286,6 +286,44 @@ Option<ContainerID> parseCgroupPath(
     const std::string& cgroupsRoot,
     const std::string& cgroup);
 
+// All cgroups v2 paths are either leaf or non-leaf paths. Leaf paths end
+// in `/leaf`, contain processes, and may impose resource constraints. Non-leaf
+// paths do not contain processes and may impose resource constraints.
+// This is done to avoid the restrictions of the "Internal Process Contraint".
+// More information: https://docs.kernel.org/admin-guide/cgroup-v2.html#no-internal-process-constraint
+//
+// The cgroup2 file system has the structure:
+// <MOUNT>/<root>/agent                    Mesos Agent constraints
+// <MOUNT>/<root>/agent/leaf               Mesos Agent process
+//
+// <MOUNT>/<root>/<id>                     Container constraints
+// <MOUNT>/<root>/<id>/leaf                Container process
+//
+// <MOUNT>/<root>/<id>/mesos/<id2>         Nested container constraints
+// <MOUNT>/<root>/<id>/mesos/<id2>/leaf    Nested container process
+//
+// For every new level of nesting, `/mesos/<idN>` is added to the path.
+// Note: We assume these nested containers are not configured to "share"
+//       cgroups with their parent, in which case they would use the same cgroup
+//       paths as their parent.
```

Comment:
can probably remove this? I think in v2 the share_cgroups flag should probably be aliased to a new name like shared_isolation, since:

* this is what the user cares about, and
* creating the cgroups lets us get stats across the executor, nested containers, even though they aren't isolated from each other, which is nice